dsl_parser/tasks.py

```diff
@@ -60,12 +60,34 @@ def _set_plan_inputs(plan, inputs=None):
                     exceptions.ERROR_INVALID_CHARS,
                     'Illegal characters in input: {0}. '
                     'Only valid ascii chars are supported.'.format(input_name))
+        elif DEFAULT in input_def and input_def[DEFAULT] is not None:
+            inputs[input_name] = input_def[DEFAULT]
         else:
-            if DEFAULT in input_def and input_def[DEFAULT] is not None:
-                inputs[input_name] = input_def[DEFAULT]
-            else:
+            # Try to get some defaults from the data_type maybe or in another
+            # words just try to parse the value before validating it.
+            try:
```

Comment:
Does this mean that if someone provides a default value we don't check the type for it?
Maybe put a couple of unit(ish?) tests in to confirm the different behaviours of this?

Reply:
No, we still do validate/verify inputs (see lines (92++).  This addition is only to make sure we try to retrieve/parse value from `data_type` first (and not report input as missing).

Reply:
Excellent,